lib/internal/event_target.js

```diff
@@ -59,7 +60,8 @@ const kBubbles = Symbol('bubbles');
 const kComposed = Symbol('composed');
 const kPropagationStopped = Symbol('propagationStopped');
 
-const isTrusted = () => false;
+const isTrustedFalse = () => false;
+const isTrustedTrue = () => true;
```

Comment:
This should use a `SafeWeakSet`, so that the `isTrusted` getter function between a trusted and untrusted event is the same:
```js
Object.getOwnPropertyDescriptor(trustedEvent, "isTrusted").get === Object.getOwnPropertyDescriptor(untrustedEvent, 'isTrusted').get;
```

```suggestion
const isTrustedSet = new SafeWeakSet();

const isTrusted = ObjectGetOwnPropertyDescriptor({
  get isTrusted() {
    return isTrustedSet.has(this);
  }
}, 'isTrusted').get;
```

Reply:
Please see Anna's comment above - this would introduce a (rather big) performance regression. The initial implementation was a bit like this (except without a WeakSet which IIRC incurs its own separate performance penalty in 8).

That said - that's a very good point (regarding wanting the reference to be the same for false/true cases.

Reply:
Hmm, @addaleax I would like to (if possible) make the issue @ExE-Boss raised work, namely:

```js
Object.getOwnPropertyDescriptor(trustedEvent, "isTrusted").get === Object.getOwnPropertyDescriptor(untrustedEvent, 'isTrusted').get;
```

Any clever trick or idea for getting the reference of different functions to equal without creating an additional closure for every event target or running into the "have a non-configurable own accessor property on an object without the overhead of creating the dictionary for slow properties" thing?

Reply:
Hm … I think the above suggestion by @ExE-Boss might work rather well, actually? If we always use the same getter, then performance-wise we’re good, and `SafeWeakSet` should give us the guarantees we’d want here in terms of unforgeability

Reply:
I'll try

Reply:
Oh, I think I misunderstood the solution - this is actually pretty clever (creating the temp object and getting the reference to avoid the different property.

Reply:
Benchmarks look good.